.github/workflows/publish.yaml

```diff
@@ -0,0 +1,56 @@
+# Creates and publishes assets for a new release.
+# To manually trigger this workflow, create a release in the GitHub UI.
+
+name: "Publish binaries"
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish-github:
+    strategy:
+      # Allowing jobs to fail until 'node-pre-gyp-github' supports failing gracefully if file already exists
+      # (https://github.com/bchr02/node-pre-gyp-github/issues/42)
+      fail-fast: false
+      matrix:
+        node_version: [12, 14, 16, 17]
+        experimental: [false]
+        system:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+        include:
+          - node_version: 16
+            system:
+              os: macos-latest
+              target: aarch64-apple-darwin
+            experimental: true
+    runs-on: ${{ matrix.system.os }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.release.tag_name }}
```

Comment:
this part is unnecessary, it will always checkout current commit with a tag by default